### ISSUE TITLE
Fluentbit keeps storage chunks busy, until the point we lose input logs, without notice.

### ISSUE THREAD

**Issue by RicharddeJong** [fluent/fluent-bit#9115]
## Bug Report

**Describe the bug**

### Situation:
We have a Fluentbit acting as a logcollector. It receives logs over UDP. It has 2 outputs and only one filter to copy the records for the second output.

When one of the 2 outputs is down, the number of storage_chunks, input_storage_chunks and input_storage_chunks_busy steadily increases. storage_chunks_up remains low.

Once the input_storage_chunks_busy reaches 2032-2048, the storage_chunks_up starts increasing suddenly, unil storage.max_chunks_up is almost reached, currently configured as 1024.

At that point the number of input_records_total stops increasing, meaning also the output to the remaining output stops. So we are losing all new data.

The amount of storage_chunks and input_storage_chunks remains constant at the point of 2048 + storage.max_chunks_up.

There is no sign, message or log record indicating that the input is paused, but the effect is that we lose our input. No new data is accepted. With tcpdump we still see udp messages coming in, they are just silently ignored.

Once we recover the output that was down, things start to flush, but in our case the default value of 128 for max_chunks_up is far too low. When we set it to 512 or 1024 the backlog is flushed at high speed. If it is too low, it seems that Fluentbit is too busy with  processing new inputs to clear out the backlog. The effect is that input records remains at a very low volume, and backlog is also not shrinking.

### Observation (possible problem):
Chunks destined for an output that is currently unavailable are never put "down" but remain "busy", so they stay in memory? That means there is less and less room for the chunks for the reamining output.

### Second situation, now only with one output. Same behaviour:
When the output is down, the number of storage_chunks, input_storage_chunks and input_storage_chunks_busy steadily increases. storage_chunks_up remains low.

Once the input_storage_chunks_busy reaches 2032-2048, the storage_chunks_up starts increasing suddenly, unil storage.max_chunks_up is almost reached.

At that point the number of input records drops to almost zero.. So we are losing all new data.

The amount of storage_chunks and input_storage_chunks remains constant at the point of 2048 + storage.max_chunks_up, so 3072 in our case.

There is no sign, message or log record indicating that the input is paused, but the effect is that we lose our input. No new data is accepted.

Below what we see in logs only retries with increasing retries, no other message. And the result of a kill -CONT to show counters:

```
[2024/07/19 16:07:33] [ info] [task] re-schedule retry=0x7f3a062278d0 1014 in the next 98 seconds
[2024/07/19 16:07:33] [ info] [task] re-schedule retry=0x7f3a0622e220 1767 in the next 222 seconds
[2024/07/19 16:07:33] [ info] [task] re-schedule retry=0x7f3a061b94e0 175 in the next 210 seconds
[2024/07/19 16:07:33] [ info] [task] re-schedule retry=0x7f3a0623a2f0 1118 in the next 163 seconds
[2024/07/19 16:07:33] [ info] [task] re-schedule retry=0x7f3a06226c50 918 in the next 63 seconds
[2024/07/19 16:07:33] [ info] [task] re-schedule retry=0x7f3a061bb0b0 380 in the next 195 seconds
[2024/07/19 16:07:33] [ info] [task] re-schedule retry=0x7f3a06227498 995 in the next 91 seconds
[2024/07/19 16:07:34] [engine] caught signal (SIGCONT)
[2024/07/19 16:07:34] Fluent Bit Dump

===== Input =====
udp_in (syslog)
│
├─ status
│  └─ overlimit     : no
│     ├─ mem size   : 2.0G (2097551459 bytes)
│     └─ mem limit  : 0b (0 bytes)
│
├─ tasks
│  ├─ total tasks   : 244
│  ├─ new           : 0
│  ├─ running       : 244
│  └─ size          : 402.5M (422034357 bytes)
│
└─ chunks
   └─ total chunks  : 1310
      ├─ up chunks  : 1025
      ├─ down chunks: 285
      └─ busy chunks: 244
         ├─ size    : 0b (0 bytes)
         └─ size err: 0

tcp_in (syslog)
│
├─ status
│  └─ overlimit     : no
│     ├─ mem size   : 0b (0 bytes)
│     └─ mem limit  : 0b (0 bytes)
│
├─ tasks
│  ├─ total tasks   : 0
│  ├─ new           : 0
│  ├─ running       : 0
│  └─ size          : 0b (0 bytes)
│
└─ chunks
   └─ total chunks  : 0
      ├─ up chunks  : 0
      ├─ down chunks: 0
      └─ busy chunks: 0
         ├─ size    : 0b (0 bytes)
         └─ size err: 0

storage_backlog.2 (storage_backlog)
│
├─ status
│  └─ overlimit     : no
│     ├─ mem size   : 0b (0 bytes)
│     └─ mem limit  : 0b (0 bytes)
│
├─ tasks
│  ├─ total tasks   : 1804
│  ├─ new           : 0
│  ├─ running       : 1804
│  └─ size          : 3.0G (3223930986 bytes)
│
└─ chunks
   └─ total chunks  : 3221
      ├─ up chunks  : 0
      ├─ down chunks: 3221
      └─ busy chunks: 1804
         ├─ size    : 0b (0 bytes)
         └─ size err: 0


===== Storage Layer =====
total chunks     : 9220
├─ mem chunks    : 0
└─ fs chunks     : 9220
   ├─ up         : 1025
   └─ down       : 8195
[2024/07/19 16:07:34] [ info] [task] re-schedule retry=0x7f3a0622bdb8 1513 in the next 534 seconds
[2024/07/19 16:07:34] [ info] [task] re-schedule retry=0x7f3a061b9440 200 in the next 385 seconds
[2024/07/19 16:07:34] [ info] [task] re-schedule retry=0x7f3a0622ff08 1997 in the next 429 seconds
[2024/07/19 16:07:34] [ info] [task] re-schedule retry=0x7f3a06227c68 1061 in the next 146 seconds
[2024/07/19 16:07:34] [ info] [task] re-schedule retry=0x7f3a06239a30 577 in the next 278 seconds
[2024/07/19 16:07:34] [ info] [task] re-schedule retry=0x7f3a0622a3a0 1318 in the next 255 seconds
[2024/07/19 16:07:34] [ info] [task] re-schedule retry=0x7f3a062296a8 1234 in the next 175 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a06226d18 965 in the next 216 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a06224dd8 709 in the next 575 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a0622d5a0 1704 in the next 51 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a0623a430 1360 in the next 179 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a0622cfb0 1635 in the next 64 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a06225490 757 in the next 64 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a061b7d20 10 in the next 176 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a0622f850 1925 in the next 78 seconds
[2024/07/19 16:07:35] [ info] [task] re-schedule retry=0x7f3a06228488 1103 in the next 217 seconds
[2024/07/19 16:07:37] [ info] [task] re-schedule retry=0x7f3a061b9d28 221 in the next 42 seconds
```

So what we see is that none if the inputs is overlimit. We do see that exaclty 2.0 GB is used in the UDP input. That aligns with 1024 storage chunks of 2MB being up. Also the sum of the busy chunks over all inputs is exactly 2048. It seems to be some magic boundary which cannot be modified with a configuration option.

Here an extract of all Prometheus counters while it was in this state:

```
fluentbit_input_bytes_total{name="udp_in"} 2657388002
fluentbit_input_records_total{name="udp_in"} 3444448
fluentbit_input_bytes_total{name="tcp_in"} 0
fluentbit_input_records_total{name="tcp_in"} 0
fluentbit_input_bytes_total{name="storage_backlog.2"} 6045397975
fluentbit_input_records_total{name="storage_backlog.2"} 7826282
fluentbit_output_proc_records_total{name="forward_logprocessor_clmp"} 0
fluentbit_output_proc_bytes_total{name="forward_logprocessor_clmp"} 0
fluentbit_output_errors_total{name="forward_logprocessor_clmp"} 0
fluentbit_output_retries_total{name="forward_logprocessor_clmp"} 13514
fluentbit_output_retries_failed_total{name="forward_logprocessor_clmp"} 0
fluentbit_output_dropped_records_total{name="forward_logprocessor_clmp"} 0
fluentbit_output_retried_records_total{name="forward_logprocessor_clmp"} 31167796
fluentbit_process_start_time_seconds{hostname="pa-traffic-fluentbit-877767875-qfw6n"} 1721397014
fluentbit_build_info{hostname="pa-traffic-fluentbit-877767875-qfw6n",version="3.0.7",os="linux"} 1721397014
fluentbit_hot_reloaded_times{hostname="pa-traffic-fluentbit-877767875-qfw6n"} 0
fluentbit_storage_chunks 9246
fluentbit_storage_mem_chunks 0
fluentbit_storage_fs_chunks 9246
fluentbit_storage_fs_chunks_up 1024
fluentbit_storage_fs_chunks_down 8222
fluentbit_input_ingestion_paused{name="udp_in"} 0
fluentbit_input_storage_overlimit{name="udp_in"} 0
fluentbit_input_storage_memory_bytes{name="udp_in"} 2097551459
fluentbit_input_storage_chunks{name="udp_in"} 1335
fluentbit_input_storage_chunks_up{name="udp_in"} 1024
fluentbit_input_storage_chunks_down{name="udp_in"} 311
fluentbit_input_storage_chunks_busy{name="udp_in"} 244
fluentbit_input_storage_chunks_busy_bytes{name="udp_in"} 0
fluentbit_input_ingestion_paused{name="tcp_in"} 0
fluentbit_input_storage_overlimit{name="tcp_in"} 0
fluentbit_input_storage_memory_bytes{name="tcp_in"} 0
fluentbit_input_storage_chunks{name="tcp_in"} 0
fluentbit_input_storage_chunks_up{name="tcp_in"} 0
fluentbit_input_storage_chunks_down{name="tcp_in"} 0
fluentbit_input_storage_chunks_busy{name="tcp_in"} 0
fluentbit_input_storage_chunks_busy_bytes{name="tcp_in"} 0
fluentbit_input_ingestion_paused{name="storage_backlog.2"} 0
fluentbit_input_storage_overlimit{name="storage_backlog.2"} 0
fluentbit_input_storage_memory_bytes{name="storage_backlog.2"} 0
fluentbit_input_storage_chunks{name="storage_backlog.2"} 3221
fluentbit_input_storage_chunks_up{name="storage_backlog.2"} 0
fluentbit_input_storage_chunks_down{name="storage_backlog.2"} 3221
fluentbit_input_storage_chunks_busy{name="storage_backlog.2"} 1804
fluentbit_input_storage_chunks_busy_bytes{name="storage_backlog.2"} 0
fluentbit_output_upstream_total_connections{name="forward_logprocessor_clmp"} 0
fluentbit_output_upstream_busy_connections{name="forward_logprocessor_clmp"} 0
fluentbit_output_chunk_available_capacity_percent{name="forward_logprocessor_clmp"} 98.309766400000001
```

The sum of all fluentbit_input_storage_chunks_busy counters is 2048. It seems to be some magical boundary which we cannot relate to any confiuration setting...

The config at this point is as follows:

```
config:
  fluentbit: |-
    service:
      daemon: off
      flush: 1
      log_level: debug
      storage.metrics: on
      storage.path: /buffer/${HOSTNAME}
      storage.sync: full
      storage.checksum: on
      storage.max_chunks_up: 1024
      storage.backlog.mem_limit: 50M
      http_server: on
      http_listen: 0.0.0.0
      http_port: 2020
      health_check: on            # Health check on curl -s http://127.0.0.1:2020/api/v1/health
      hc_errors_count: 5          # The error count to meet the unhealthy requirement, this is a sum for all output plugins in a defined HC_Period
      hc_retry_failure_count: 5   # The retry failure count to meet the unhealthy requirement, this is a sum for all output plugins in a defined HC_Period
      hc_period: 60               # The time period by second to count the error and retry failure data point
      parsers_file: parsers.conf

    pipeline:
      inputs:
        - name: syslog
          mode: udp
          listen: 0.0.0.0
          port: 514
          log_level: info
          parser: BSD-syslog-rfc3164
          # buffer_chunk_size: 1M
          # buffer_max_size: 4M
          receive_buffer_size: 26214400
          storage.type: filesystem
          tag: log.PA.Traffic
          alias: udp_in

        - name: syslog
          mode: tcp
          listen: 0.0.0.0
          port: 514
          log_level: info
          parser: BSD-syslog-rfc3164
          # buffer_chunk_size: 1M
          # buffer_max_size: 4M
          storage.type: filesystem
          tag: log.PA.Traffic
          alias: tcp_in

      outputs:
        - name: forward
          match: log.*
          upstream: upstream.conf
          retry_limit: no_limits
          require_ack_response: true
          alias: forward_logprocessor_clmp
          workers: 4
          storage.total_limit_size: 512G

  parsers: |-
    [PARSER]
      Name BSD-syslog-rfc3164
      Format regex
      # AE Message corrupt Regex /^\<(?<pri>[0-9]+)\>(?<time>[A-Za-z]+ {1,2}[^ ]* [^ ]*) (?<host>[^ ]*) *(?<message>.*)$/
      Regex /^\<(?<pri>[0-9]+)\>(?<time>[^ ]* {1,2}[^ ]* [^ ]*) (?<host>[^ ]*) *(?<message>.*)$/

  upstream: |-
    [UPSTREAM]
      name forward-balancing
    [NODE]
      name fluentd-kubernetes
      host fluentd-clmp-dta.logcollector.svc.cluster.local
      port 24224
```

### New test with storage.backlog.mem_limit: 3G
Behavior identical.

No sign of input pause:

```
[2024/07/19 17:11:14] [debug] [output:forward:forward_logprocessor_clmp] request 1085902 bytes to flush
[2024/07/19 17:11:14] [debug] [net] socket #1151 could not connect to 172.30.129.51:24224
[2024/07/19 17:11:14] [debug] [net] could not connect to fluentd-clmp-dta.logcollector.svc.cluster.local:24224
[2024/07/19 17:11:14] [debug] [upstream] connection #-1 failed to fluentd-clmp-dta.logcollector.svc.cluster.local:24224
[2024/07/19 17:11:14] [error] [output:forward:forward_logprocessor_clmp] no upstream connections available
[2024/07/19 17:11:14] [debug] [out flush] cb_destroy coro_id=4046
[2024/07/19 17:11:14] [debug] [output:forward:forward_logprocessor_clmp] task_id=1305 assigned to thread #2
[2024/07/19 17:11:14] [debug] [output:forward:forward_logprocessor_clmp] request 2048034 bytes to flush
[2024/07/19 17:11:14] [debug] [output:forward:forward_logprocessor_clmp] task_id=14 assigned to thread #3
[2024/07/19 17:11:14] [debug] [output:forward:forward_logprocessor_clmp] request 710694 bytes to flush
[2024/07/19 17:11:14] [debug] [net] socket #1092 could not connect to 172.30.129.51:24224
[2024/07/19 17:11:14] [debug] [net] could not connect to fluentd-clmp-dta.logcollector.svc.cluster.local:24224
[2024/07/19 17:11:14] [debug] [retry] re-using retry for task_id=756 attempts=8
[2024/07/19 17:11:14] [debug] [upstream] connection #-1 failed to fluentd-clmp-dta.logcollector.svc.cluster.local:24224
[2024/07/19 17:11:14] [error] [output:forward:forward_logprocessor_clmp] no upstream connections available
[2024/07/19 17:11:14] [debug] [out flush] cb_destroy coro_id=4046
[2024/07/19 17:11:14] [ warn] [engine] failed to flush chunk '1-1721401014.747391636.flb', retry in 1119 seconds: task_id=756, input=udp_in > output=forward_logprocessor_clmp (out_id=0)
[2024/07/19 17:11:14] [debug] [retry] re-using retry for task_id=1305 attempts=8
[2024/07/19 17:11:14] [ warn] [engine] failed to flush chunk '1-1721401194.968129565.flb', retry in 1198 seconds: task_id=1305, input=udp_in > output=forward_logprocessor_clmp (out_id=0)
[2024/07/19 17:11:14] [debug] [net] socket #1088 could not connect to 172.30.129.51:24224
[2024/07/19 17:11:14] [debug] [net] could not connect to fluentd-clmp-dta.logcollector.svc.cluster.local:24224
[2024/07/19 17:11:14] [debug] [retry] re-using retry for task_id=14 attempts=9
[2024/07/19 17:11:14] [debug] [upstream] connection #-1 failed to fluentd-clmp-dta.logcollector.svc.cluster.local:24224
[2024/07/19 17:11:14] [error] [output:forward:forward_logprocessor_clmp] no upstream connections available
[2024/07/19 17:11:14] [debug] [out flush] cb_destroy coro_id=4046
[2024/07/19 17:11:14] [ warn] [engine] failed to flush chunk '1-1721400769.783957787.flb', retry in 1309 seconds: task_id=14, input=udp_in > output=forward_logprocessor_clmp (out_id=0)
[2024/07/19 17:11:15] [debug] [output:forward:forward_logprocessor_clmp] task_id=853 assigned to thread #0
[2024/07/19 17:11:15] [ info] [task] re-schedule retry=0x7f6f1308aa60 385 in the next 209 seconds
[2024/07/19 17:11:15] [debug] [output:forward:forward_logprocessor_clmp] request 1642090 bytes to flush
[2024/07/19 17:11:15] [ info] [task] re-schedule retry=0x7f6f131b4fc0 817 in the next 201 seconds
[2024/07/19 17:11:15] [ info] [task] re-schedule retry=0x7f6f13060908 1838 in the next 489 seconds
[2024/07/19 17:11:15] [ info] [task] re-schedule retry=0x7f6f1305f1e8 1611 in the next 11 seconds
[2024/07/19 17:11:15] [debug] [net] socket #1115 could not connect to 172.30.129.51:24224
[2024/07/19 17:11:15] [debug] [net] could not connect to fluentd-clmp-dta.logcollector.svc.cluster.local:24224
[2024/07/19 17:11:15] [debug] [upstream] connection #-1 failed to fluentd-clmp-dta.logcollector.svc.cluster.local:24224
[2024/07/19 17:11:15] [error] [output:forward:forward_logprocessor_clmp] no upstream connections available
[2024/07/19 17:11:15] [debug] [out flush] cb_destroy coro_id=4047
[2024/07/19 17:11:15] [debug] [retry] re-using retry for task_id=853 attempts=8
[2024/07/19 17:11:15] [ warn] [engine] failed to flush chunk '1-1721401045.656180236.flb', retry in 765 seconds: task_id=853, input=udp_in > output=forward_logprocessor_clmp (out_id=0)
[2024/07/19 17:11:16] [ info] [task] re-schedule retry=0x7f6f0a28d170 1135 in the next 1001 seconds
[2024/07/19 17:11:16] [ info] [task] re-schedule retry=0x7f6f1305f148 1676 in the next 57 seconds
[2024/07/19 17:11:16] [ info] [task] re-schedule retry=0x7f6f0a28c0b8 1029 in the next 311 seconds
[2024/07/19 17:11:17] [ info] [task] re-schedule retry=0x7f6f13088aa8 168 in the next 1183 seconds
```
Messages like the last are repeated over and over.

kill -CONT:

```
[2024/07/19 17:13:29] [ info] [task] re-schedule retry=0x7f6f1305d6e0 1532 in the next 410 seconds
[2024/07/19 17:13:29] [ info] [task] re-schedule retry=0x7f6f13061560 1853 in the next 164 seconds
[2024/07/19 17:13:30] [engine] caught signal (SIGCONT)
[2024/07/19 17:13:30] Fluent Bit Dump

===== Input =====
udp_in (syslog)
│
├─ status
│  └─ overlimit     : no
│     ├─ mem size   : 2.0G (2097548675 bytes)
│     └─ mem limit  : 0b (0 bytes)
│
├─ tasks
│  ├─ total tasks   : 2048
│  ├─ new           : 0
│  ├─ running       : 2048
│  └─ size          : 3.3G (3517369719 bytes)
│
└─ chunks
   └─ total chunks  : 3084
      ├─ up chunks  : 1025
      ├─ down chunks: 2059
      └─ busy chunks: 2048
         ├─ size    : 0b (0 bytes)
         └─ size err: 0

tcp_in (syslog)
│
├─ status
│  └─ overlimit     : no
│     ├─ mem size   : 0b (0 bytes)
│     └─ mem limit  : 0b (0 bytes)
│
├─ tasks
│  ├─ total tasks   : 0
│  ├─ new           : 0
      ├─ up chunks  : 0
      ├─ down chunks: 0
      └─ busy chunks: 0
         ├─ size    : 0b (0 bytes)
         └─ size err: 0

storage_backlog.2 (storage_backlog)
│
├─ status
│  └─ overlimit     : no
│     ├─ mem size   : 0b (0 bytes)
│     └─ mem limit  : 0b (0 bytes)
│
├─ tasks
│  ├─ total tasks   : 0
│  ├─ new           : 0
│  ├─ running       : 0
│  └─ size          : 0b (0 bytes)
│
└─ chunks
   └─ total chunks  : 0
      ├─ up chunks  : 0
      ├─ down chunks: 0
      └─ busy chunks: 0
         ├─ size    : 0b (0 bytes)
         └─ size err: 0


===== Storage Layer =====
total chunks     : 3084
├─ mem chunks    : 0
└─ fs chunks     : 3084
   ├─ up         : 1025
   └─ down       : 2059
[2024/07/19 17:13:30] [ info] [task] re-schedule retry=0x7f6f1305df28 1491 in the next 471 seconds
[2024/07/19 17:13:30] [ info] [task] re-schedule retry=0x7f6f1305ebd0 1649 in the next 256 seconds
[2024/07/19 17:13:30] [ info] [task] re-schedule retry=0x7f6f1305d848 1459 in the next 469 seconds
```

All metrics:

```
root@pa-traffic-fluentbit-5bd975b694-qdrr6:~# curl -q http://localhost:2020/api/v2/metrics/prometheus |grep ^fl
fluentbit_uptime{hostname="pa-traffic-fluentbit-5bd975b694-qdrr6"} 2026
fluentbit_input_bytes_total{name="udp_in"} 6124194419
fluentbit_input_records_total{name="udp_in"} 8034502
fluentbit_input_bytes_total{name="tcp_in"} 0
fluentbit_input_records_total{name="tcp_in"} 0
fluentbit_input_bytes_total{name="storage_backlog.2"} 0
fluentbit_input_records_total{name="storage_backlog.2"} 0
fluentbit_output_proc_records_total{name="forward_logprocessor_clmp"} 478710
fluentbit_output_proc_bytes_total{name="forward_logprocessor_clmp"} 366060978
fluentbit_output_errors_total{name="forward_logprocessor_clmp"} 0
fluentbit_output_retries_total{name="forward_logprocessor_clmp"} 15974
fluentbit_output_retries_failed_total{name="forward_logprocessor_clmp"} 0
fluentbit_output_dropped_records_total{name="forward_logprocessor_clmp"} 0
fluentbit_output_retried_records_total{name="forward_logprocessor_clmp"} 35959642
fluentbit_process_start_time_seconds{hostname="pa-traffic-fluentbit-5bd975b694-qdrr6"} 1721400685
fluentbit_build_info{hostname="pa-traffic-fluentbit-5bd975b694-qdrr6",version="3.0.7",os="linux"} 1721400685
fluentbit_hot_reloaded_times{hostname="pa-traffic-fluentbit-5bd975b694-qdrr6"} 0
fluentbit_storage_chunks 3142
fluentbit_storage_mem_chunks 0
fluentbit_storage_fs_chunks 3142
fluentbit_storage_fs_chunks_up 1024
fluentbit_storage_fs_chunks_down 2118
fluentbit_input_ingestion_paused{name="udp_in"} 0
fluentbit_input_storage_overlimit{name="udp_in"} 0
fluentbit_input_storage_memory_bytes{name="udp_in"} 2097548675
fluentbit_input_storage_chunks{name="udp_in"} 3142
fluentbit_input_storage_chunks_up{name="udp_in"} 1024
fluentbit_input_storage_chunks_down{name="udp_in"} 2118
fluentbit_input_storage_chunks_busy{name="udp_in"} 2048
fluentbit_input_storage_chunks_busy_bytes{name="udp_in"} 0
fluentbit_input_ingestion_paused{name="tcp_in"} 0
fluentbit_input_storage_overlimit{name="tcp_in"} 0
fluentbit_input_storage_memory_bytes{name="tcp_in"} 0
fluentbit_input_storage_chunks{name="tcp_in"} 0
fluentbit_input_storage_chunks_up{name="tcp_in"} 0
fluentbit_input_storage_chunks_down{name="tcp_in"} 0
fluentbit_input_storage_chunks_busy{name="tcp_in"} 0
fluentbit_input_storage_chunks_busy_bytes{name="tcp_in"} 0
fluentbit_input_ingestion_paused{name="storage_backlog.2"} 0
fluentbit_input_storage_overlimit{name="storage_backlog.2"} 0
fluentbit_input_storage_memory_bytes{name="storage_backlog.2"} 0
fluentbit_input_storage_chunks{name="storage_backlog.2"} 0
fluentbit_input_storage_chunks_up{name="storage_backlog.2"} 0
fluentbit_input_storage_chunks_down{name="storage_backlog.2"} 0
fluentbit_input_storage_chunks_busy{name="storage_backlog.2"} 0
fluentbit_input_storage_chunks_busy_bytes{name="storage_backlog.2"} 0
fluentbit_output_upstream_total_connections{name="forward_logprocessor_clmp"} 0
fluentbit_output_upstream_busy_connections{name="forward_logprocessor_clmp"} 0
fluentbit_output_chunk_available_capacity_percent{name="forward_logprocessor_clmp"} 98.891636000000005
root@pa-traffic-fluentbit-5bd975b694-qdrr6:~#
```

At this point only 200 messages per minute are added to input_records_total and are added to the storage_chunks.

From the moment where storage_chunks_busy reaches 2048, all .flb files have exactly the same size, 2068480. It seems files are padded at the end to reach this number of bytes.

**To Reproduce**

- Steps to reproduce the problem:
See config above

**Expected behavior**
When output is down, buffer to disk, so create chunks that are down and continue doing so, until storage.total_limit_size is reached.

**Screenshots**

First attempt with 2 outputs:

Inputs worked well but went to 0:
![Screenshot 2024-07-19 170553](https://github.com/user-attachments/assets/dc895859-9a98-450d-994d-0efbdfca1ab9)

Storage info:
![Screenshot 2024-07-19 170613](https://github.com/user-attachments/assets/31c8fec3-d394-4cc5-b51e-d2f66067e38e)

Storage info showing storage_chunks_up increases fast when storage_chunks_busy reaches 2048:
![Screenshot 2024-07-19 170629](https://github.com/user-attachments/assets/3762e279-1a56-422d-812a-124cc0972ed8)

Second attempt with storage.backlog.mem_limit: 3G:

Inputs dropping when storage_chunks_busy at 2048 and storage_chunks_up at 1024:
![Screenshot 2024-07-19 171221](https://github.com/user-attachments/assets/cc7a760c-baf5-4485-8a1e-8ac8602bd3fd)

Storage chunks:
![Screenshot 2024-07-19 171231](https://github.com/user-attachments/assets/6a55b383-f873-4ff4-b8b8-082c87695aba)

![Screenshot 2024-07-19 171257](https://github.com/user-attachments/assets/2e77cb3a-0e2d-4f42-a618-fccbb95186f0)

**Your Environment**

* Version used: 3.0.7 and 3.1.2
* Configuration: Running in Kubernetes, receiving input via Syslog over UDP
* Environment name and version (e.g. Kubernetes? What version?): Recent Kubernetes
* Filters and plugins: see config above

**Additional context**
We are losing UDP inputs in case an output is down. We may be missing some obvious parameter, but we tried very hard to follow documentation and fiddle with settings.

**Comment by RicharddeJong** [comment]
This may be the same issue as https://github.com/fluent/fluent-bit/issues/6993

**Comment by RicharddeJong** [comment]
It also seems very similar to https://github.com/fluent/fluent-bit/issues/5217. The good news is that we can easily reproduce it since we have an input that keeps hammering until busy chunks reaches 2048 and max_chunks_up is reached.
We have `retry_limit: no_limits`, maybe that is why chunks remain busy? I cannot loose my logs, I want to be able to resend them after the weekend or whenever I recovered my output. FB just needs to store to disk so I don't loose my logs..

**Comment by leonardo-albertovich** [comment]
Hi @RicharddeJong, I have been trying to reproduce this issue for a few days but haven't been able to, particularly I can easily get the task count to the limit (which I'll add some context about tomorrow) but I couldn't manage to get the mapped chunk to grow in the way you describe.

Would you mind helping me reproduce it so I can fix it? I'd appreciate it if you talked to me in slack to have a more fluid conversation but any hints or tips you can give me would be highly appreciated.

**Comment by RicharddeJong** [comment]
Sure, please let me know how to find you for DM. I use the same name in Slack.

**Comment by leonardo-albertovich** [comment]
I'm in the public slack server and my handle is `Leonardo Almiñana`

**Comment by edsiper** [comment]
@RicharddeJong @leonardo-albertovich if the conversation happened in Slack, I would kindly ask you to post here a resolution or update please.

**Comment by leonardo-albertovich** [comment]
I'm still waiting for Richard and his coworkers to come back from their vacations so they can send me some additional resources and we can continue the investigation.

**Comment by neighbour-oldhuang** [comment]
version 3.1.5  类似问题，我的是 fluent-bit 发送日志到 es ，es那边负载很低， fluent-bit这边 cpu 一直单核心跑满 （25%used of 4Cores）
![image](https://github.com/user-attachments/assets/2a6ae0b6-e613-42b4-9597-f5cd58ebc4c7)
![image](https://github.com/user-attachments/assets/1d912a65-8ccd-4b1e-a28c-a00a99f14f4e)
以上是出现busy的配置，
![image](https://github.com/user-attachments/assets/c2fac5a9-a312-4a6b-9e2a-46ee651c56ef)
结果就是只有少量日志被es收集到

**Comment by tooptoop4** [comment]
still an issue
